source/includes/pipelines/_07-compare.md.erb

```diff
@@ -0,0 +1,115 @@
+## Compare pipeline instances
+
+```shell
+
+$ curl 'http://ci.example.com/go/api/pipelines/pipeline1/compare/1/3' \
+       -u 'username:password' \
+       -H 'Accept: <%= data.apis.versions.latest %>'
+```
+
+> The above command returns JSON structured like this:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+```
+
+```json
+{
+  "_links" : {
+    "self" : {
+      "href" : "http://ci.example.com/go/api/pipelines/:pipeline_name/compare/:from_counter/:to_counter"
+    },
+    "doc" : {
+      "href" : "https://api.gocd.org/current/#compare-pipeline-instances"
+    }
+  },
+  "pipeline_name" : "pipeline1",
+  "from_counter" : 1,
+  "to_counter" : 3,
+  "is_bisect" : false,
+  "changes" : [ {
+    "material" : {
+      "type" : "git",
+      "attributes" : {
+        "destination" : null,
+        "filter" : null,
+        "invert_filter" : false,
+        "name" : null,
+        "auto_update" : true,
+        "url" : "git@github.com:sample_repo/example.git",
+        "branch" : "master",
+        "submodule_folder" : null,
+        "shallow_clone" : false
+      }
+    },
+    "revision" : [ {
+      "revision_sha" : "some-random-sha",
+      "modified_by" : "username <username@github.com>",
+      "modified_at" : "2019-10-15T12:32:37Z",
+      "commit_message" : "some commit message"
+    } ]
+  },  {
+    "material" : {
+      "type" : "dependency",
+      "attributes" : {
+        "pipeline" : "upstream",
+        "stage" : "upstream_stage",
+        "name" : "upstream_material",
+        "auto_update" : true
+      }
+    },
+    "revision" : [ {
+      "revision" : "upstream/1/upstream_stage/1",
+      "pipeline_counter" : "1",
+      "completed_at" : "2019-10-17T06:55:07Z"
+    } ]
+  } ]
+}
+```
+
+The compare pipeline API allows users to find the difference in the materials between the two instance.
+
+<%= available_since('19.10.0') %>
+
+<p class='http-request-heading'>HTTP Request</p>
+
+`GET /go/api/pipelines/:pipeline_name/compare/:from_counter/:to_counter`
+
+<p class='http-request-return-description'>Returns</p>
+
+The [comparsion result object](#the-comparison-result-object).
+
+<%=
+describe_object 'The comparison result object', since: '19.10.0' do
+ pipeline_name 'String', 'The name of the pipeline.'
+ from_counter  'Integer', 'The instance count which is being compared'
+ to_counter    'Integer', 'The instance count to which the above instance is compared'
+ is_bisect     'Boolean', 'States whether or not either one of the instance is a bisect. Meaning, the instance counter was trigggered with an older revision.'
```

Comment:
@arvindsv Is this correct? Does it need more explanation?

Reply:
@kritika-singh3 It's correct. However, since we  haven't ever explained bisection, maybe we can clarify it a bit? Something like: "States whether or not either one of the instances is a bisection. If true, it means that the instances are not directly comparable, since at least one of them was manually triggered with older revisions of materials and the comparison might not make sense because of the chosen revisions."

Reply:
This makes more sense. Thanks!